File: playground/ALaCarteDemo/ALaCarteDemo/generated-output/src/ALaCarteWorkspace/projects/src/app/dashboard-cards/dashboard-card.ts
// Copyright (c) Quinntyne Brown. All Rights Reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

export class Options {
  public top: number;
  public left: number;
  public height: number;
  public width: number;
}

export class DashboardCard {
  public dashboardCardId: number;
  public name: string;
  public cardId: number;
  public dashboardId: number;
  public options: Options = new Options();
}

File: playground/ALaCarteDemo/ALaCarteDemo/generated-output/src/Commitments.Core/Model/DashboardCardAggregate/DashboardCard.cs
// Copyright (c) Quinntyne Brown. All Rights Reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using Commitments.Core.Model.CardAggregate;
using Commitments.Core.Model.CardLayoutAggregate;
using Commitments.Core.Model.DashboardAggregate;
using Newtonsoft.Json.Linq;
using System.ComponentModel.DataAnnotations.Schema;

namespace Commitments.Core.Model.DashboardCardAggregate;

public class DashboardCard
{
    public Guid DashboardCardId { get; set; }
    [ForeignKey("Dashboard")]
    public Guid DashboardId { get; set; }
    [ForeignKey("Card")]
    public Guid CardId { get; set; }
    [ForeignKey("CardLayout")]
    public Guid CardLayoutId { get; set; }
    public Dashboard Dashboard { get; set; }
    public Card Card { get; set; }
    public CardLayout CardLayout { get; set; }
    public JObject Options { get; set; }
}